### ISSUE TITLE
Deploying to GitHub fails with git error

### ISSUE THREAD

**Issue by ccrisan** [JamesIves/github-pages-deploy-action#33]
**Describe the bug**

Deploying to GitHub Pages fails at the end with:

    Deploying to GitHub...
    1/2 (0)
    2/2 (1)
    error: src refspec ***@github.com/myusername/myrepo.git does not match any.
    error: failed to push some refs to 'https://'
    ##[error]Docker run failed with exit code 1

**Reproduce**

My (simplified) workflow looks something like this:

    name: CI
    on: push
    jobs:
      jsdoc:
        name: Publish Docs
        runs-on: ubuntu-latest
        steps:
          - uses: actions/checkout@master
          - uses: JamesIves/github-pages-deploy-action@master
            env:
              ACCESS_TOKEN: ${{ secrets.GH_PAGES_ACCESS_TOKEN }}
              BASE_BRANCH: dev
              BRANCH: gh-pages
              FOLDER: docs
              BUILD_SCRIPT: npm install && npx jsdoc -c jsdoc.conf.json

**Expected behavior**

The workflow step should succeed by publishing the `docs` folder to GitHub Pages.

**Comment by JamesIves** [comment]
Is this a private or public repo, and is the `dev` branch published? It sounds to me like it can't find the branch for whatever reason.

**Comment by ccrisan** [comment]
It's a public repo and yes, the `dev` branch is there. [This](https://github.com/qtoggle/qui/) is the repo and [this](https://github.com/qtoggle/qui/blob/github-workflows/.github/workflows/ci.yml) is the workflow file.

**Comment by JamesIves** [comment]
Which permissions did you get the access token? I'm wondering if the permissions are not sufficient which is why it's throwing the error as it attempts to commit:  https://github.com/JamesIves/github-pages-deploy-action/blob/master/entrypoint.sh#L86-L90

**Comment by ccrisan** [comment]
Only these: 

![image](https://user-images.githubusercontent.com/76639/66130153-ebf79700-e5f9-11e9-81d4-ba5788fb872a.png)

It didn't seem like anything else would be required for pushing to the repo. Would it?

**Comment by JamesIves** [comment]
My test repo has these permissions - not sure if that will fix it but it's worth giving it a shot.

<img width="781" alt="Screen Shot 2019-10-03 at 9 25 32 AM" src="https://user-images.githubusercontent.com/10888441/66130563-e5e4c480-e5bf-11e9-838e-8cd9a8e2af63.png">

**Comment by ccrisan** [comment]
@JamesIves you were right about not having the correct permissions. In fact, I did two separate mistakes:

1. I (probably) pasted the wrong access token value into the secret variable, most likely including extra spaces/tabs or being incomplete. Once I've got it right, the user (owning the token) has been correctly identified.

2. The user (a CI/CD dedicated user, owning the aforementioned token) was part of the team but had no write access. Once I explicitly granted her write access to the repo, the first four repo permissions were enough for the action to succeed.

While this issue is definitely invalid, I believe that insufficient user permissions should somehow be indicated by the action's script. Maybe an explicit, separate test command to push something to the repo could help identifying the lack of privileges.
